stdlib/string.mli

```diff
@@ -316,6 +316,14 @@ val equal: t -> t -> bool
 (** The equal function for strings.
     @since 4.03.0 *)
 
+val is_prefix : t -> t -> bool
+(** [String.is_prefix s pre] tests if [pre] is a prefix of [s]
```

Comment:
As far as I'm concerned this is the wrong argument order. If I read `String.is_prefix s0 s1` I expect the function to check that `s0` is a prefix of `s1`. It's also an inconvenient order for partial application.

Containers' `is_prefix` has the order I suggest. Batteries and base do not have this order but I don't think it's a good idea not follow them on that one. 

The same holds for `is_suffix` below.

Reply:
Actually the current variant is consistent with the existing `check_suffix` function from `Filename` and consistent with the other search and check functions in the `String` module which all take the string on which to perform the actions are as first argument.

Reply:
I think we need at least one label here if we want the function, so that the order cannot be confused.

Reply:
> Actually the current variant is consistent with the existing `check_suffix` function from `Filename`

I don't think there's any need for consistency with that one. The name is sufficently different. 

> `String` module which all take the string on which to perform the actions are as first argument.

There's already `split_on_char` that does not follow this. Following what you mention results in something that reads wrong if applied unlabelled, please don't do this. Code reading is more important than the consistency (which never really exists in OCaml) you are looking for.